File: C++/Programmers/Level1/09 소수 만들기.cpp
#include <vector>
#include <iostream>
using namespace std;

int solution(vector<int> nums) {
    int answer = 0;
    int prime[3001] = {0, };

    prime[0] = 1;
    prime[1] = 1;

    for(int i=2; i<3001; i++){
        if(prime[i]==0){
            for(int j=i*2; j<3001; j+=i){
                prime[j] = 1;
            }
        }
    }

    for(int i=0; i<nums.size(); i++){
        for(int j=i+1; j<nums.size(); j++){
            for(int k=j+1; k<nums.size(); k++){
                if(prime[nums[i]+nums[j]+nums[k]]==0) {
                    answer++;
                }
            }
        }
    }

    return answer;
}

File: JS/Programmers/Level1/09 소수 만들기.js
function solution(nums) {
    var answer = 0;
    var prime = new Array(3001);

    prime[0] = 1;
    prime[1] = 1;

    for(var i=2; i<3001; i++){
        if(prime[i] == undefined){
            for(j = i*2; j<3001; j+=i){
                prime[j] = 1;
            }
        }
    }

    for(var i=0; i<nums.length; i++){
        for(var j=i+1; j<nums.length; j++){
            for(var k=j+1; k<nums.length; k++){
                if(prime[nums[i]+nums[j]+nums[k]] == undefined)
                    answer++;
            }
        }
    }

    return answer;
}